presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -360,6 +373,92 @@ protected RelationType visitShowColumns(ShowColumns showColumns, AnalysisContext
     }
 
     @Override
+    protected RelationType visitShowCreateTable(ShowCreateTable showCreateTable, AnalysisContext context)
+    {
+        QualifiedObjectName tableName = createQualifiedObjectName(session, showCreateTable, showCreateTable.getName());
+
+        if (metadata.getView(session, tableName).isPresent()) {
+            throw new SemanticException(NOT_SUPPORTED, showCreateTable, "'%s' is a view, not a table", tableName);
+        }
+
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+        if (!tableHandle.isPresent()) {
+            throw new SemanticException(MISSING_TABLE, showCreateTable, "Table '%s' does not exist", tableName);
+        }
+
+        ConnectorTableMetadata connectorTableMetadata = metadata.getTableMetadata(session, tableHandle.get()).getMetadata();
+
+        List<TableElement> columns = connectorTableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .map(column -> new TableElement(column.getName(), column.getType().getDisplayName()))
+                .collect(toImmutableList());
+
+        Map<String, Object> properties = connectorTableMetadata.getProperties();
+        Map<String, PropertyMetadata<?>> allTableProperties = metadata.getTablePropertyManager().getAllTableProperties().get(tableName.getCatalogName());
+        Map<String, Expression> sqlProperties = new HashMap<>();
+
+        for (Map.Entry<String, Object> propertyEntry : properties.entrySet()) {
+            String propertyName = propertyEntry.getKey();
+            Object value = propertyEntry.getValue();
+            if (value == null) {
+                throw new PrestoException(INTERNAL_ERROR, String.format("Property %s for table %s cannot have a null value", propertyName, tableName));
+            }
+
+            if (value instanceof OptionalInt) {
```

Comment:
This shouldn't be needed. The property value should already be converted from `OptionalInt` to the `Integer` value. If any connector is returning `OptionalInt`, the encoder needs to be fixed.